example/passthrough_helpers.h

```diff
@@ -23,11 +23,55 @@
  * SUCH DAMAGE
  */
 
+#ifndef FUSE_EXAMPLE_PASSTHROUGH_HELPERS_H_
+#define FUSE_EXAMPLE_PASSTHROUGH_HELPERS_H_
+
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#ifdef __FreeBSD__
+#include <sys/socket.h>
+#include <sys/un.h>
+#endif
+
+static inline int do_fallocate(int fd, int mode, off_t offset, off_t length)
+{
+#ifdef HAVE_FALLOCATE
+	if (fallocate(fd, mode, offset, length) == -1)
+		return -errno;
+	return 0;
+#else  // HAVE_FALLOCATE
+
+#ifdef HAVE_POSIX_FALLOCATE
+	if (mode == 0)
+		return -posix_fallocate(fd, offset, length);
+#endif
+
+#ifdef HAVE_FSPACECTL
+	// 0x3 == FALLOC_FL_KEEP_SIZE | FALLOC_FL_PUNCH_HOLE
```

Comment:
Why don't you actually include the right headers so you can use these symbols, instead of hard coding their values here?

Reply:
Is there a fuse_kernel.h for BSD and we could use that? Maybe we should better rename fuse_kernel.h to fuse_linux_kernel.h and have it system specific? Because we already have an issue with OSX (https://github.com/libfuse/libfuse/issues/1075)